### DISCUSSION TITLE
Imposing normal boundary conditions

### DISCUSSION THREAD

**Discussion by raphaelouillon** [CliMA/Oceananigans.jl#2610] [Computational science!]
Hi all,

I have seen this question discussed here and there but I am not sure if the answer has changed with more recent updates to the code: Is there a reasonable way to impose an inflow (normal flow) boundary condition? If I understand correctly, there isn't a real outflow boundary condition, but imposing the normal flow velocity at the inlet and outlet, and adding a sponge layer at the outlet can do an ok job. Do we have an example of such a problem somewhere? A good old flow past a cylinder for instance? 

Thanks for your thoughts on this!

**Comment by glwagner** [comment]
It's not really supported, but you can try to use `OpenBoundaryCondition(1)`, for example. You should write some validation tests to see if this works and report results here.

However, since we have to impose a sponge layer anyways, it may make more sense to use `Periodic` boundaries and impose the inflow via the sponge layer at the "outlet". This works for simple / periodic inflow+outflow conditions.

**Reply by raphaelouillon** [reply]
Thanks @glwagner , the sponge approach seems to be working well!
